cmd/podman/common/create_opts.go

```diff
@@ -401,7 +414,7 @@ func ContainerCreateToContainerCLIOpts(cc handlers.CreateContainerConfig) (*Cont
 		cliOpts.Restart = policy
 	}
 
-	if cc.HostConfig.MemorySwappiness != nil {
+	if cc.HostConfig.MemorySwappiness != nil && !rootless.IsRootless() {
```

Comment:
this also needs to account for cgroupv2.

The condition should be: `rootless.IsRootless() && cgroupsv2 && cgroupDriver == "systemd'`